### PR TITLE
[Move] Introduce freeze/unfreeze move scripts

### PR DESCRIPTION
**Pull request by moeziniaf**
## Motivation

Freeze/Unfreeze scripts for AOS - move functions already exist to freeze txn operations from an account

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

cargo test -- freeze_unfreeze_account
Transaction_builder.rs freeze/unfreeze test
Tests account can not execute txns once frozen.

### CODE REVIEW

File: language/stdlib/src/transaction_scripts.rs [hunk 1]
```diff
@@ -37,6 +37,7 @@ pub enum StdlibScript {
     CreateAccount,
     CreateEmptyAccount,
     EmptyScript,
+    Freeze,
```
Comment (runtian-zhou) [inline language/stdlib/src/transaction_scripts.rs:40]: FreezeAccount?

Comment (sblackshear) [inline language/stdlib/src/transaction_scripts.rs:40]: Nit: `FreezeAccount` would be slightly more explicit

**Comment by andll** [comment]
Can we also add UnfreezeAccount script in same PR?

**Comment by sblackshear** [comment]
Would you mind adding tests for each new script [here](https://github.com/libra/libra/blob/master/language/e2e-tests/src/tests/transaction_builder.rs) to make sure the arguments are type-correct + the script does what we expect?

File: language/stdlib/src/transaction_scripts.rs (hunk 1 above)
Comment (tzakian) [reply language/stdlib/src/transaction_scripts.rs]: Not entirely sure why `stdlib.mv` would need to be updated with this. That should only change if there are changes to the stdlib modules unless I'm mistaken (and I don't see any changes to them).

**Comment by tzakian** [comment]
I don't think there should be a staged `stdlib.mv` under transaction scripts, (can happen if you run it from the wrong directory). You'll probably want to remove it and do a `cargo run` from `language/stdlib`

File: language/stdlib/transaction_scripts/freeze_account.move [hunk 2]
```diff
@@ -0,0 +1,7 @@
+script {
+use 0x0::LibraAccount;
+// Script for freezing account by authorized initiator
```
Comment (sblackshear) [inline language/stdlib/transaction_scripts/freeze_account.move:3]: Nit: align comment with `fun main`

File: language/e2e-tests/src/tests/transaction_builder.rs [hunk 3]
```diff
@@ -65,6 +67,47 @@ fn register_preburn_burn() {
...
+    // Grant freezing privilege to assocation account
+    executor.execute_and_apply(association.signed_script_txn(
+        encode_grant_association_privilege(lbr_ty, *association.address()),
+        2,
+    ));
+    // Execute freeze on account
+    executor.execute_and_apply(
+        association.signed_script_txn(encode_freeze_account(*account.address()), 3),
+    );
+
```
Comment (sblackshear) [inline language/e2e-tests/src/tests/transaction_builder.rs:104]: Not strictly necessary, but could be useful to test that sending a tx from `account` fails after freeze and succeeds after unfreeze.

Comment (sblackshear) [reply language/e2e-tests/src/tests/transaction_builder.rs:104]: Any transaction will do for testing this, but key rotation is probably the easiest: https://github.com/libra/libra/blob/master/language/e2e-tests/src/tests/rotate_key.rs

**Comment by moeziniaf** [comment]
Included relevant more robust test that frozen account can not perform txns

**Pull request closed without merging**